homeassistant/components/arcticspa/__init__.py

```diff
@@ -0,0 +1,51 @@
+"""The Arctic Spa integration."""
+from __future__ import annotations
+
+import logging
+
+from pyarcticspas import Spa
+from pyarcticspas.error import SpaHTTPException, TooManyRequestsError, UnauthorizedError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import ArcticSpaDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up an Arctic Spa from a config entry."""
+
+    device = Spa(entry.data[CONF_API_KEY])
+
+    # The API has no login endpoint. We will use a status check to test access and availability.
+    try:
+        await device.async_status()
+    except UnauthorizedError as ex:
+        raise ConfigEntryAuthFailed("Invalid API token") from ex
```

Comment:
When raising this error you need to have a reauth flow. Since adding this would be a bit too big for this PR, raise a ConfigEntryError instead. (This way it won't retry again).

You can implement reauth in a followup

Reply:
Thanks! I've submitted this in a new commit.